lib/house.rb

```diff
@@ -0,0 +1,65 @@
+class House
+  attr_reader :address, :rooms
+
+  def initialize(price_paramater, address_parameter)
+    @price = price_paramater
+    @address = address_parameter
+    @rooms = []
+  end
+
+  def price
+    @price.gsub("$", '').to_i
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    price > 500000 ? true : false
+  end
+
+  def rooms_from_category(category)
+    @rooms.select {|room| room.category == category}
+  end
+
+  def area
+    total_area = 0
+    @rooms.each {|room| total_area += room.area}
+    total_area
+  end
+
+  def details
+    details = {"price" => price, "address" => @address}
+    details
+  end
+
+  def price_per_square_foot
+    total_per_square_foot = 0
+    if area != 0
+      total_per_square_foot = (price / area.to_f).round(2)
+    end
+    total_per_square_foot
+  end
+
+  def rooms_sorted_by_area
+    rooms_sorted = []
+    rooms.each do |room|
+      if rooms_sorted.first == nil
+        rooms_sorted.append room
+      elsif room.area > rooms_sorted.first.area
+        rooms_sorted.prepend room
+      else
+        rooms_sorted.append room
+      end
+    end
+    rooms_sorted
```

Comment:
This doesn't quite work... which is understandable, as there's a whole universe of algorithmic challenges around building sort methods... fortunately, you don't have to worry about that. Ruby has a nice, built-in sort method for you. (defaults to ascending order, least to greatest)

```
def rooms_sorted_by_area
   @rooms.sort_by {|room| room.area}.reverse
end
```
(reversed to switch the order to descending, as per the interaction pattern)